code/scripting/api/objs/enums.h

```diff
@@ -127,6 +127,10 @@ const int32_t LE_SCORE_FICTION_VIEWER        = 114;
 const int32_t LE_NOT_YET_PRESENT             = 115; // the sequence and offsets of these three #defines should correspond to the ShipStatus enums
 const int32_t LE_PRESENT                     = 116;
 const int32_t LE_EXITED                      = 117;
+const int32_t LE_DC_IS_HULL                  = (1<<0); // the following definitions are bitfields independent of the usual indexing
+const int32_t LE_DC_VAPORIZE                 = (1<<1);
+const int32_t LE_DC_SET_VELOCITY             = (1<<2);
+const int32_t LE_DC_FIRE_HOOK                = (1<<3);
```

Comment:
This will cause problems.
For one, these Enums will compare equal to other enums with the same index, which is not what we want.
Secondly, the getName function of these enums will return the first enum whose index is equal to their index, which is also not the correct name.
Thirdly, there is currently no API operator that would allow a user to combine two of these creation flags.

Right now, the global enum class is not suitable for these kind of bitfield enums. It would first need a refactor to be capable of holding multiple enums with the same value, but different indices. 

A _very_ ugly quick fix would be to modify the old enum indices in such a way, that neither 1, 2, 4 or 8 exist apart from these new bitfield values, but that jeopardizes anything which might have looked at the actual value of the Enum, as well as pretty much requiring a quick followup with a refactor to enums.

Reply:
There are API operators that allow one to combine enums... they are EnumAND and EnumOR.  They were originally created for the OS_ definitions that are used for object sounds.  Those enums supposedly already have the equality and getName issues you mention, but in practice this hasn't been a problem, at least with the scripts they have been used in.

Yeah, a refactor to enum flags might be desirable.  I'm not sure offhand how to do that though, except maybe by just duplicating the existing enum code and tweaking it for bitfields.

Reply:
Oh god, it's horrible. I see.
Well...
Keep it for now then, I have an idea how to improve that and will do so after this PR